dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/security/config/AuthenticationListener.java

```diff
@@ -137,7 +137,14 @@ private void registerSuccessfulLogin( String username )
         if ( Objects.nonNull( user ) && !readOnly )
         {
             user.updateLastLogin();
-            userService.updateUser( user );
+            try
+            {
+                userService.updateUser( user );
+            }
+            catch ( Exception e )
+            {
+                log.warn( "Failed to update the user!", e );
```

Comment:
We should make a note of this and check the logs to see what the exception is. A user should be updated with last logged in info, so if it's not it would be a functional bug.

Reply:
This is due to the fact that the OAuth2 client/user is not a real user and hence the lookup will fail. So in any other cases this should not happen. Consider this a hack, since there is no other reliable way to know what user type it is right now. This problem will disappear when we remove the old OAuth2 support or we do a bigger refactor to include User "type" information into the User "object".